pact/namespaces/ns.pact

```diff
@@ -31,25 +27,48 @@
     (enforce (is-charset CHARSET_LATIN1 name)
              "Name must be in latin1 charset"))
 
+  (defun create-principal-namespace:string
+      ( g:guard
+        )
+    " Format principal namespace as Pact hash (BLAKE2b256) of principal \
+    \ in hex truncated to 160 bits (40 characters), prepended with 'n_'.\
+    \ Only w: and k: account protocols are supported. "
+
+    (let
+      ((ty (typeof-principal (create-principal g))))
+
+      ;; only w: and k: currently supported
+      (if (or (= ty "k:") (= ty "w:"))
+        (+ "n_" (take 40 (int-to-str 16 (str-to-int 64 (hash g)))))
+        (enforce false
+          (format "Unsupported guard protocol: {}" [ty]))
+        ))
+  )
+
   (defun validate:bool
       ( ns-name:string
         ns-admin:guard
         )
-    " Manages namespace install for Chainweb. Requires active row in registry \
+    " Manages namespace install for Chainweb. \
+    \ Allows principal namespaces. \
+    \ Non-principal namespaces require active row in registry \
     \ for NS-NAME with guard matching NS-ADMIN."
 
-    (validate-name ns-name)
+    (if (= (create-principal-namespace ns-admin) ns-name)
+
+      true ;; allow principal namespaces
 
-    (with-default-read registry ns-name
-      { 'admin-guard : ns-admin
-      , 'active : false }
-      { 'admin-guard := ag
-      , 'active := is-active }
+      (with-default-read registry ns-name       ;; otherwise enforce registry
+        { 'admin-guard : ns-admin
+        , 'active : false }
+        { 'admin-guard := ag
+        , 'active := is-active }
 
         (enforce is-active "Inactive or unregistered namespace")
         (enforce (= ns-admin ag) "Admin guard must match guard in registry")
 
-        true))
+        true)
+      ))
```

Comment:
Do we want to prevent principal namespaces from being registered via `write-registry`? Per `validate` they "do no harm" as the principal would not hit the record, so it's more correctness for it's own sake.

Reply:
So, we did this in [previous commits](https://github.com/kadena-io/chainweb-node/blob/e21f1d3620efc00b67563143928b52401991615b/pact/namespaces/v2/ns.pact#L75) but it was removed on the basis of being unnecessary. I'm fine with adding it back using the new-and-improved principal ns format. It's no sweat over here, we just need to commit to something.

Reply:
No, if we removed it let's stick with that. Resolving